homeassistant/components/synology_dsm/__init__.py

```diff
@@ -63,6 +63,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass.config_entries.async_update_entry(
             entry, data={**entry.data, CONF_VERIFY_SSL: DEFAULT_VERIFY_SSL}
         )
+    if (options := dict(entry.options)) and options.get(CONF_TIMEOUT):
+        options.pop(CONF_TIMEOUT)
```

Comment:
Would be slightly faster if you only converted it to a `dict` if it was going to be updated